tratto/src/main/java/star/tratto/util/FileUtils.java

```diff
@@ -0,0 +1,124 @@
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+
+/**
+ * I/O utilities: creating files, writing to files, retrieving all Java files, etc.
+ */
+public class FileUtils {
+    /**
+     * Combines given information into a file path.
+     *
+     * @param dirPath a base root path
+     * @param fileName the file name
+     * @param fileFormat the type of file (e.g. ".txt", ".json")
+     * @param projectName the corresponding project being analyzed
+     * @return a file path which combines the given information into
+     * format:
+     *  "[dirPath]/[projectName]/[fileName].[fileFormat.getExtension()]"
+     */
+    public static Path getPath(
+            String dirPath,
+            String fileName,
+            FileFormat fileFormat,
+            String projectName
+    ) {
+        return Paths.get(dirPath, projectName, fileName + fileFormat.getExtension());
+    }
+
+    /**
+     * Creates an empty file at a given path. Creates parent directories if
+     * necessary. If the file already exists, then this method does nothing.
+     *
+     * @param path a file
+     * @throws IOException if an error occurs while creating the parent
+     * directories or new file
+     */
+    public static void createFile(Path path) throws IOException {
+        Files.createDirectories(path.getParent());
+        if (!Files.exists(path)) {
+            Files.createFile(path);
+        }
+    }
+
+    /**
+     * Writes {@code content} to {@code path}. Creates a new file/directories
+     * if necessary. Overwrites any previous content.
+     *
+     * @param path a file
+     * @param content an object to be mapped as JSON content
+     * @throws IOException if unable to create files/directories or unable to
+     * write content to file
+     */
+    public static void write(Path path, Object content) throws IOException {
+        createFile(path);
+        ObjectMapper objectMapper = new ObjectMapper();
+        objectMapper.writerWithDefaultPrettyPrinter().writeValue(path.toFile(), content);
+    }
+
+    /**
+     * Gets all Java files in a given directory (and subdirectories).
+     *
+     * @param dir a directory
+     * @return all Java files (as Path objects) in {@code dir}
+     * @throws IOException if unable to collect files from {@code dir}
+     */
+    public static List<Path> getAllJavaFilesFromDirectory(Path dir) throws IOException {
+        try (Stream<Path> walk = Files.walk(dir)) {
+            return walk
+                    .filter(p -> p.getFileName().toString().endsWith(".java"))
+                    .collect(Collectors.toList());
+        } catch (IOException e) {
+            // catch exception to avoid resource leak.
+            throw new IOException(String.format("Error in collecting all files from %s.%n", dir), e);
+        }
+    }
+
+    /**
+     * Returns an absolute file path.
```

Comment:
This only returns an absolute path if `dirPath` is already absolute.  If `dirPath` is relative, the returned result is relative.  Perhaps you need to call `toAbsolutePath()`.